osu.Game/IO/Serialization/Converters/TypedListConverter.cs

```diff
@@ -14,7 +14,7 @@ namespace osu.Game.IO.Serialization.Converters
     /// reconstruct the objects with their original types.
     /// </summary>
     /// <typeparam name="T">The type of objects contained in the <see cref="List{T}"/> this attribute is attached to.</typeparam>
-    public class TypedListConverter<T> : JsonConverter
+    public class TypedListConverter<T> : JsonConverter<List<T>>
```

Comment:
`IList<T>` will fix the CI errors here (covariance).

Reply:
In fact it doesn't fix the error.
I'm just obeying the type check in `CanConvert`. It seems to be violated somewhere.

Reply:
Actually `IEnumerable<T>` fixes it. Inspired by the `(IEnumerable<T>)value` cast in `WriteJson`.